File: C++/MaxMinWODnC.cpp
#include<iostream>
using namespace std;

void min_max(int *array, int length)  {
    int maximum = array[0];
    int minimum = array[0];
    
    for(int index=1 ; index<length ; index++)   {
        if(array[index] > maximum)  {
            maximum = array[index];
        }
        else if(array[index] < minimum)   {
            minimum = array[index];
        }
    }

    cout<<"Maximum value: "<<maximum<<endl<<" Minimum value: "<<minimum;
}

int main()  {
    cout<<"Enter the length of array: ";
    int length;
    cin>>length;
    cout<<"Enter the array: ";
    int array[length];

    for(int index=0 ; index<length ; index++)    {
        cin>>array[index];
    }

    min_max(array, length);

    return 0;
}

File: Java/MaxMinWODnC.java
import java.awt.*;
import java.util.Scanner;

public class MaxMinWODnC {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter length of the array: ");
        byte len = scanner.nextByte();
        System.out.print("Enter array elements: ");
        int[] arr = new int[len];
        for (byte index=0 ; index<len ; index++)    {
            arr[index] = scanner.nextByte();
        }

        Point answer  = min_max(arr, len);

        System.out.println("Minimum value is: " + (int)answer.getX() + "\nMaximum value is: " + (int)answer.getY());
    }

    private static Point min_max(int[] arr, byte len)   {
        Point minMax = new Point();

        for (byte index=0 ; index< len ; index++)   {
            if(arr[index] < minMax.getX())  {
                minMax.x = arr[index];
            }
            else if(arr[index] > minMax.getY()) {
                minMax.y = arr[index];
            }
        }

        return minMax;
    }
}
